x-pack/plugins/uptime/common/rules/uptime_rule_field_map.ts

```diff
@@ -0,0 +1,78 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export const uptimeRuleFieldMap = {
+  // common fields
+  'monitor.id': {
+    type: 'keyword',
+  },
+  'url.full': {
+    type: 'keyword',
+  },
+  reason: {
+    type: 'text',
+  },
+  // monitor status alert fields
+  'error.message': {
+    type: 'text',
+  },
+  'agent.name': {
+    type: 'keyword',
+  },
+  'monitor.name': {
+    type: 'keyword',
+  },
+  'monitor.type': {
+    type: 'keyword',
+  },
+  'observer.geo.name': {
+    type: 'keyword',
+  },
+  // tls alert fields
+  'cert_status.count': {
+    type: 'integer',
+  },
+  'cert_status.aging_count': {
+    type: 'integer',
+  },
+  'cert_status.aging_common_name_and_date': {
```

Comment:
i am bit confused here, cert_status is plural or singular here. if it's a status of multiple certs , that means `aging_common_name_and_date` is probably an array?

Reply:
i personally think may be this confusion is coming out of because our cert status alert still acts as a bulk alert message, where it sends a curated message for all expiring certificates, i think we should refactor that to work as alert instance per expiring certificate just like what we have for monitor status. any thoughts @justinkambic ?

Reply:
Yes, you are right about the batched/bulk alert message. The intention here was to story the message string in this field. My strategy in this first go around for indexing alert documents was to index all the same things we are keeping in alert state. We store the bulk alert message as a concatenated string of all the relevant information in alert state and this is meant to match here.